wp-includes/rest-api/endpoints/class-wp-rest-customize-sections-controller.php

```diff
@@ -0,0 +1,305 @@
+<?php
+/**
+ * REST API: WP_REST_Customize_Sections_Controller class
+ *
+ * @package WordPress
+ * @subpackage REST_API
+ * @since ?.?.?
+ */
+
+/**
+ * Core class to access sections via the REST API.
+ *
+ * @since ?.?.?
+ *
+ * @see WP_REST_Controller
+ */
+class WP_REST_Customize_Sections_Controller extends WP_REST_Controller {
+
+	/**
+	 * Constructor.
+	 *
+	 * @since 4.7.0
+	 * @access public
+	 */
+	public function __construct() {
+		$this->namespace = 'customize/v1';
+		$this->rest_base = 'sections';
+	}
+
+	/**
+	 * Ensure customize manager.
+	 *
+	 * @return WP_Customize_Manager Manager.
+	 * @global WP_Customize_Manager $wp_customize
+	 */
+	public function ensure_customize_manager() {
+		global $wp_customize;
+		if ( empty( $wp_customize ) ) {
+			$wp_customize = new WP_Customize_Manager(); // WPCS: global override ok.
+		}
+		if ( ! did_action( 'customize_register' ) ) {
+
+			/** This action is documented in wp-includes/class-wp-customize-manager.php */
+			do_action( 'customize_register', $wp_customize );
+		}
+		$wp_customize->prepare_controls();
```

Comment:
@westonruter Added the `prepare_controls` here since this will "fill in" the controls of a section. Do you think it's better to create a separate logic to detect the related controls instead? Seems that some `get_item()` tests with custom sections are failing when using `prepare_controls`. I haven't yet looked into that more thoroughly to detect the issue but wanted to ask your opinion about using `prepare_controls` like this here first.